exporters/failure/collector_jira.py

```diff
@@ -17,15 +35,36 @@ def __init__(self, user, apikey, server, projects):
         super().__init__(server, user, apikey)
         self.projects = projects
 
-    def search_issues(self):
-        options = {"server": self.server}
+    def _connect_to_jira(self) -> JIRA:
+        """Method to connect to JIRA instance which may be cloud based
+        or self-hosted.
+        """
+        jira_client = None
+        server = self.server
+        email = self.user
+        api_key = self.apikey
+
         # Connect to Jira
-        jira = JIRA(options, basic_auth=(self.user, self.apikey))
+        jira_client = JIRA(options={"server": server}, basic_auth=(email, api_key))
+
+        # Ensure connection was performed
+        try:
+            jira_client.session()
+        except JIRAError as error:
+            logging.error(
+                "Status: %s, Error Response: %s", error.status_code, error.text
+            )
+            raise error
```

Comment:
Using `raise` without the `error` will re-throw the exception while keeping the original traceback correct